### ISSUE TITLE
Error: `Unsupported parameters for (inventory_source) module: update_on_project_update`

### ISSUE THREAD

**Issue by nodje** [redhat-cop/automate_tower_genie_collections#399]
I get this error when trying to activate the option on an otherwise working Inventory Source config. (Using the devel branch from 2 weeks ago)

Though I see the param listed in the role README.md and in `controller_configuration/roles/inventory_sources/tasks/main.yml`

I'm not sure what's wrong here.

Here's the Tower (3.8.5) error report for the task `TASK [redhat_cop.controller_configuration.inventory_sources : Configure Inventory Source | Wait for finish the Inventory Source creation]`:
```
{
    "started": 1,
    "finished": 1,
    "stdout": "",
    "stderr": "/usr/lib/python3.6/site-packages/OpenSSL/crypto.py:12: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.\n  from cryptography import x509\n",
    "stdout_lines": [],
    "stderr_lines": [
        "/usr/lib/python3.6/site-packages/OpenSSL/crypto.py:12: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.",
        "  from cryptography import x509"
    ],
    "ansible_job_id": "289356536929.2264922",
    "results_file": "/tmp/.ansible_async/289356536929.2264922",
    "msg": "Unsupported parameters for (inventory_source) module: update_on_project_update. Supported parameters include: controller_host (tower_host), notification_templates_error, name, notification_templates_started, source_vars, host_filter, overwrite, new_name, description, validate_certs (tower_verify_ssl), controller_username (tower_username), source_path, notification_templates_success, enabled_value, timeout, credential, inventory, organization, verbosity, controller_config_file (tower_config_file), enabled_var, custom_virtualenv, controller_oauthtoken (tower_oauthtoken), source, state, controller_password (tower_password), source_project, overwrite_vars, update_on_launch, execution_environment, update_cache_timeout.",
    "invocation": {
        "module_args": {
            "name": "Inventory Source for Developpement",
            "inventory": "ANSIBLE Inventory for Developpement Env",
            "source": "scm",
            "source_path": "by_ENV.yml",
            "source_vars": {
                "ENV": "DEVELOPPEMENT"
            },
            "overwrite": true,
            "overwrite_vars": true,
            "custom_virtualenv": "/apps/tower/custom-venv/plugin/",
            "source_project": "ANSIBLE plugin source project",
            "update_on_project_update": true,
            "state": "present",
            "controller_host": "https://automation.staging.intra",
            "controller_username": "",
            "controller_password": "",
            "validate_certs": false,
            "controller_oauthtoken": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "controller_config_file": null,
            "new_name": null,
            "description": null,
            "enabled_var": null,
            "enabled_value": null,
            "host_filter": null,
            "credential": null,
            "execution_environment": null,
            "organization": null,
            "timeout": null,
            "verbosity": null,
            "update_on_launch": null,
            "update_cache_timeout": null,
            "notification_templates_started": null,
            "notification_templates_success": null,
            "notification_templates_error": null
        }
    },
    "_ansible_no_log": false,
    "attempts": 1,
    "changed": false,
    "__inventory_source_job_async_results_item": {
        "failed": 0,
        "started": 1,
        "finished": 0,
        "ansible_job_id": "289356536929.2264922",
        "results_file": "/tmp/.ansible_async/289356536929.2264922",
        "changed": false,
        "__controller_source_item": {
            "name": "Inventory Source for Developpement",
            "description": "",
            "source": "scm",
            "source_project": "ANSIBLE plugin source project",
            "source_path": "by_ENV.yml",
            "source_vars": {
                "ENV": "DEVELOPPEMENT"
            },
            "inventory": "ANSIBLE Inventory for Developpement Env",
            "update_on_project_update": true,
            "overwrite": true,
            "overwrite_vars": true,
            "custom_virtualenv": "/apps/tower/custom-venv/plugin/"
        },
        "ansible_loop_var": "__controller_source_item"
    },
    "ansible_loop_var": "__inventory_source_job_async_results_item",
    "_ansible_item_label": {
        "failed": 0,
        "started": 1,
        "finished": 0,
        "ansible_job_id": "289356536929.2264922",
        "results_file": "/tmp/.ansible_async/289356536929.2264922",
        "changed": false,
        "__controller_source_item": {
            "name": "Inventory Source for Developpement",
            "description": "",
            "source": "scm",
            "source_project": "ANSIBLE plugin source project",
            "source_path": "by_ENV.yml",
            "source_vars": {
                "ENV": "DEVELOPPEMENT"
            },
            "inventory": "ANSIBLE Inventory for Developpement Env",
            "update_on_project_update": true,
            "overwrite": true,
            "overwrite_vars": true,
            "custom_virtualenv": "/apps/tower/custom-venv/plugin/"
        },
        "ansible_loop_var": "__controller_source_item"
    }
}
```

**Comment by djdanielsson** [comment]
the issue is you are using tower 3.8.5 not aap 2.2 which is what the latest version of this collection is built for

**Comment by adonisgarciac** [comment]
Hi, I have tested it on tower 3.8.6 without problem.

**Comment by djdanielsson** [comment]
that is good to know, I do know that some parts do not work on Tower anymore but I currently do not have a list of what does not.... we probably should make that. but anyways I double checked the output you gave and I do not see an error anywhere. can you please re-try and give the full output of the error along with your inventory source list that you are trying to build? @nodje

**Comment by nodje** [comment]
It'd be surprising to see this option, which is similar on Tower and AAP2, stop working on Tower.

Here is the source for the inventory source:
```
  - name: "Inventory Source for Developpement"
    description: ""
    source: scm
    source_project: "{{ orgs }} plugin source project"
    source_path: "by_ENV.yml"
    source_vars:
      ENV: "DEVELOPPEMENT"
    inventory: "{{ orgs }} Inventory for Developpement Env"
    update_on_project_update: true
    overwrite: true
    overwrite_vars: true
    custom_virtualenv: "/apps/tower/custom-venv/plugin/"
```    
The output I posted above is the entire content (redacted) of the failed task.
I believe I cannot provide anything more complete than that.

The error message is :
```
"msg": "Unsupported parameters for (inventory_source) module: update_on_project_update. Supported parameters include: controller_host (tower_host), notification_templates_error, name, notification_templates_started, source_vars, host_filter, overwrite, new_name, description, validate_certs (tower_verify_ssl), controller_username (tower_username), source_path, notification_templates_success, enabled_value, timeout, credential, inventory, organization, verbosity, controller_config_file (tower_config_file), enabled_var, custom_virtualenv, controller_oauthtoken (tower_oauthtoken), source, state, controller_password (tower_password), source_project, overwrite_vars, update_on_launch, execution_environment, update_cache_timeout."
```

The parameter `update_on_project_update` is not part of the list of supported parameters.

**Comment by adonisgarciac** [comment]
@nodje are you still trying in v3.8.5?

**Comment by sean-m-sullivan** [comment]
I've been looking at this, the current awx.awx collection, and other things. This seems to be a side effect of this
https://github.com/ansible/awx/pull/12366
Looking at latest awx this has been removed , I would check either the awx.awx version you are using or the ansible.controller you are using. Let us know which, and the version, but that is the likely culprit.

**Comment by nodje** [comment]
Yes, I'm still trying on 3.8.5 and preparing a test on AAP2.2 in parallel.

I'm using ansible-controller-4.2.1
(I've transitioned from awx.awx-21.6.0 to it since I'm using Tower and AAP but it was equally working as far as aI can tell)

**Comment by adonisgarciac** [comment]
Hi @nodje, as @sean-m-sullivan told, update_on_project_update was removed from awx.awx collection. Where you are using awx instance, I think you can use awx.awx collection in the same version that awx server. But to work properly against tower or controller, I suggest you to use ansible.tower or ansible.controller.

If you check both collections, you can see that update_on_project_update is still present. But if you see awx.awx documentation, you can see that update_on_project_update was removed:

AWX.AWX:  https://docs.ansible.com/ansible/latest/collections/awx/awx/inventory_source_module.html
ANSIBLE.TOWER: https://console.redhat.com/ansible/automation-hub/repo/published/ansible/tower/content/module/tower_inventory_source_update
ANSIBLE.CONTROLLER: https://console.redhat.com/ansible/automation-hub/repo/published/ansible/controller/content/module/inventory_source

**Comment by nodje** [comment]
@adonisgarciac I'm not using awx.awx.

I'm using ansible.controller-4.2.1 and redhat_cop.controller_configuration.
And I can find the parameter `update_on_project_update` in the collection sources.

The collections used by my project are included in the Git project sources as we are in an offline env, and we do not have an AutomationHub at our disposal yet.

That makes it even harder to understand this error message.

**Comment by adonisgarciac** [comment]
Ok, If you are against tower, you should use ansible.tower even although it doesn't seem the problem

**Comment by nodje** [comment]
I've changed the collections included in my project so that I use ansible.tower along with red_cop.controller_configuration.

But I keep having the issue.

I realized on the way that I keep having the following warning:
```
"warnings": [
        "You are using the awx version of this collection but connecting to Red Hat Ansible Tower"
    ],
``` 
I didn't pay attention to these warnings at the begining because I was using awx.awx collection.
But I can't explain why these warning keeps coming up even thought the awx.awx collection is not available.

At least that seems like a plausible explanation to my issue since as @adonisgarciac mentioned, the AWX.AWX https://docs.ansible.com/ansible/latest/collections/awx/awx/inventory_source_module.html doesn't contain `update_on_project_update` anymore.
But I don't see how my project is using awx.awx collection...

BTW, as awx.awx is the upstream for Tower/AAP shall we expect ` update_on_project_update` to disappear from ansible.tower/ansible.controller as well? Why would this option disappear? It's just really useful for some scenarios.

**Comment by djdanielsson** [comment]
yes we do expect the latest versions of ansible.tower and ansible.controller to no longer have `update_on_project_update`, you would have to go put an issue in with awx.awx to find out why they decided to remove the option but we are limited to what those modules provide so we will be removing the option as well to prevent the error from happening.

**Comment by sean-m-sullivan** [comment]
It was removed from the AWX/Tower itself, the modules just facilitate accessing it, so I would refer to the linked awx issue about it of why they removed, but we are just conduits of the API

**Comment by nodje** [comment]
I still see the option on Tower 3.8.5 and AAP2.2 but so be it.
